internal/bminventory/inventory.go

```diff
@@ -668,8 +669,13 @@ func (b *bareMetalInventory) RegisterClusterInternal(ctx context.Context, kubeKe
 		return nil, common.NewApiError(http.StatusBadRequest, err)
 	}
 
+	var mirroredRegistries []string
+	if mirrorRegistryConfiguration != nil {
+		mirroredRegistries = extractMirroredRegistriesFromConfig(log, mirrorRegistryConfiguration)
```

Comment:
What happens if extraction fails, should we handle this error?

Reply:
Thanks for bringing this up! I think if extraction fails, we should definitely have a log for it (which is done in the `extractMirroredRegistriesFromConfig`). The mirror registry validation is done in another part and should be handled there, should I leave a comment about that here?

Main reason is that I don't think registering a cluster and validating its pull secret should fail at this point because the mirror registry is incorrect. What do you think?

Reply:
We should just put a comment here to acknowledge that we are happy this is handled elsewhere

Reply:
Added the comment below, thanks!